src/ForkCMS/Utility/Thumbnails.php

```diff
@@ -0,0 +1,160 @@
+        $folders = $this->getFolders($inPath);
+
+        // loop folders
+        foreach ($folders as $folder) {
+            $this->generateThumbnail($folder, $forSourceFile);
+        }
+    }
+
+    private function generateThumbnail(array $folder, string $forSourceFile): void
+    {
+        /** @var ImageInterface $image */
+        $image = $this->imagine->open($forSourceFile);
+
+        /** @var Box */
+        $box = $image->getSize();
+
+        // if the width & height are specified we should ignore the aspect ratio
+        if ($folder['width'] !== null && $folder['height'] !== null) {
+            // we scale on the smaller dimension
+            if ($box->getWidth() > $box->getHeight()) {
+                $width  = $box->getWidth() * ($folder['height']/$box->getHeight());
+                $height =  $folder['height'];
+
+                // we center the crop in relation to the width
+                $cropPoint = new Point(max($width - $folder['width'], 0)/2, 0);
+            } else {
+                $width  = $folder['width'];
+                $height =  $box->getHeight() * ($folder['width']/$box->getWidth());
+
+                // we center the crop in relation to the height
+                $cropPoint = new Point(0, max($height - $folder['height'], 0)/2);
+            }
+
+            // we scale the image to make the smaller dimension fit our resize box
+            $image = $image->thumbnail(new Box($width, $height), ImageInterface::THUMBNAIL_OUTBOUND);
+
+            // and crop exactly to the box
+            $image->crop($cropPoint, new Box($folder['width'], $folder['height']));
+        } else {
+            // redefine box because we need to calculate box size
+            $box = ($folder['width'] !== null) ? $box->widen($folder['width']) : $box->heighten($folder['height']);
+
+            // we use resize and not thumbnail, because thumbnail has memory leaks
+            $image->resize($box);
+        }
+
+        $image->save($folder['path'] . '/' . basename($forSourceFile));
+    }
+
+    /**
+     * Get the folders
+     *
+     * @param string $inPath The path
+     * @param bool $includeSourceFolder Should the source-folder be included in the return-array.
+     *
+     * @return array
+     */
+    public function getFolders(string $inPath, bool $includeSourceFolder = false): array
+    {
+        if (!$this->filesystem->exists($inPath)) {
+            return [];
+        }
+
+        $folders = [];
+        $nameFilter = ($includeSourceFolder) ? 'source' : '/^([0-9]*)x([0-9]*)$/';
+
+        foreach ($this->finder->directories()->in($inPath)->name($nameFilter)->depth('== 0') as $directory) {
+            $dirname = $directory->getBasename();
+            $chunks = explode('x', $dirname, 2);
+
+            if (!$includeSourceFolder && count($chunks) !== 2) {
+                continue;
+            }
+
+            $width = (($dirname !== 'source') && ($chunks[0] !== '')) ? (int) $chunks[0] : null;
```

Comment:
woulnd't it be better to just check here if the chunk is numerical?

Reply:
Indeed, changed to `is_int(xxx)`.
Fixed by d1281a4